vic/drivers/classic/src/alloc_veg_hist.c

```diff
@@ -38,29 +38,27 @@ alloc_veg_hist(int                nrecs,
 {
     int i, j;
 
-    (*veg_hist) = (veg_hist_struct **) calloc(nrecs, sizeof(veg_hist_struct *));
+    (*veg_hist) = calloc(nrecs, sizeof(*(*veg_hist)));
     if ((*veg_hist) == NULL) {
         log_err("Memory allocation error in alloc_veg_hist().");
     }
 
     for (i = 0; i < nrecs; i++) {
-        (*veg_hist)[i] = (veg_hist_struct *) calloc(nveg,
-                                                    sizeof(veg_hist_struct));
+        (*veg_hist)[i] = calloc(nveg, sizeof(*(*veg_hist)[i]));
```

Comment:
OK, looking a little more closely, I see that now...

On Fri, Aug 28, 2015 at 5:32 PM, Bart Nijssen <notifications@github.com>
wrote:

> In vic/drivers/classic/src/alloc_veg_hist.c
> <https://github.com/UW-Hydro/VIC/pull/257#discussion_r38256292>:
>
> >      if ((*veg_hist) == NULL) {
> >          log_err("Memory allocation error in alloc_veg_hist().");
> >      }
> >
> >      for (i = 0; i < nrecs; i++) {
> > -        (*veg_hist)[i] = (veg_hist_struct *) calloc(nveg,
> > -                                                    sizeof(veg_hist_struct));
> > +        (*veg_hist)[i] = calloc(nveg, sizeof(*(*veg_hist)[i]));
>
> Not at all
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/UW-Hydro/VIC/pull/257/files#r38256292>.
>